doc/Overview/Mvux/Advanced/Messaging.md

```diff
@@ -0,0 +1,162 @@
+        });
+    }
+
+    public void MessageReceived(MyMessage myMessage)
+    {
+        ...
+    }
+}
+```
+
+The `MyMessage` type is defined in a location accessible to both the sender and the receiver:
+
+```csharp
+public record MyMessage(string Message);
+```
+
+Then, in another location or module in the app, a `MyMessage` is sent:
+
+```csharp
+using CommunityToolkit.Mvvm.Messaging;
+
+public partial record AnotherModel
+{
+    public AnotherModel(IMessenger messenger)
+    {
+        this.Messenger = messenger;
+    }
+
+    protected IMessenger Messenger { get; }
+
+    public IFeed<string> Message => ...
+
+    public void SendMessage(string message)
+    {
+        var myMessenge = new MyMessage(message);
+        Messenger.Send(myMessage);
+    }
+}
+```
+
+When the messege is sent using the `Send` method, it's intercepted using the registration in the `MyModel` and its `MessageReceived` method is invoked along with the `MyMessage` parameter.
+
+## MVUX integration with the messenger
+
+MVUX harnesses the power of the Community Toolkit messenger and adds extension methods that enable you to listen to entity changes received from the messenger, and have them automatically applied to the state or list-state storing the entities in the current model. The current entity-change types are supported: created, updated, deleted.
+For instance, if there is a command in the model that when executed creates a new entity and stores it to the database using a service, the service can send a 'created' entity-change message with the messenger, which can then be intercepted in the model to have the state or list-state update itself and display the newly created entity received from the messenger.
+
+### Observe
+
+In the example below, we have a model that displays a state-list of `Person` entities received from a service, loaded using a state, with the [`Async`](xref:Overview.Mvux.ListStates#async) factory-method.
+
+As you can gather from the code, the service interacts with an external source to load and save `Person` data. In the example below we can see the use of two of its methods: `GetAllAsync`, and `CreateNameAsync`.
+
+There's also a method `CreateNewPerson` which gets generated as a command to be invoked from the View (refer to [commands](xref:Overview.Mvux.Advanced.Commands) to learn about how MVUX generates commands). This method uses `CreateRandomName`, which generates a random name. Its implementation has been removed for brevity.
+
+The line using the MVUX messaging extension method, is the one calling `messenger.Observe`. Read the code and this line will be explained thereafter.
+
+```csharp
+using CommunityToolkit.Mvvm.Messaging;
+using Uno.Extensions.Reactive;
+using Uno.Extensions.Reactive.Messaging;
+
+public record Person(string Name);
+
+public partial record PeopleModel
+{
+    protected IPeopleService PeopleService { get; }
+
+    public IListState<Person> People => ListState.Async(this, PeopleService.GetAllAsync);
```

Comment:
Does this need to be a state, or can it work with a feed?

Reply:
It needs to be a state as we're changing the number of entities in it adding newly created entities and removing deleted ones from it.
There are other `Observe` overloads which can be used with feeds which I'm still trying to figure out what would be a good scenario to document them with.